packages/fdc3-get-agent/test/support/TestServerContext.ts

```diff
@@ -46,46 +50,65 @@ export class TestServerContext implements ServerContext<ConnectionDetails> {
   }
 
   getMatchingInstance(url: string): ConnectionDetails | undefined {
-    return this.instances.find(ca => ca.url === url);
+    const details = this.instances.find(ca => ca.url === url);
+    if (!details) {
+      const knownInstances = this.instances.map(inst => {
+        return { appId: inst.appId, url: inst.url };
+      });
+      console.error(
+        'No connection instance found - will return a mismatched instance, url: ',
+        url,
+        '\nknown instances: ',
+        knownInstances
+      );
+      return this.instances[0];
+    }
+    return details;
   }
 
   async shutdown(): Promise<void> {
     await Promise.all(this.instances.map(i => i.internalPort.close()));
     await Promise.all(this.instances.map(i => i.externalPort.close()));
   }
 
+  /** Used to mock connections to the server from apps. Must be called before the app attempts to connect for that connection to succeed. */
   async open(appId: string): Promise<InstanceID> {
-    const ni = this.nextInstanceId++;
-    if (appId.includes('missing')) {
+    const url = dummyInstanceDetails.find(value => value.appId === appId)?.url;
+    if (!url) {
+      console.error('TestServerContext Tried to open an unknown appId');
       throw new Error(OpenError.AppNotFound);
     } else {
-      const mc = new MessageChannel();
-      const internalPort = mc.port1;
-      const externalPort = mc.port2;
-
-      (internalPort as any).name = 'internalPort-' + ni;
-      (externalPort as any).name = 'externalPort-' + ni;
-
-      internalPort.start();
-
-      const connectionDetails = {
-        appId,
-        instanceId: 'uuid-' + ni,
-        connected: false,
-        connectionId: 'uuid-' + ni,
-        externalPort,
-        internalPort,
-        url: 'https://dummyOrigin.test/path',
-        state: State.Pending,
-      };
-
-      this.instances.push(connectionDetails);
-      internalPort.onmessage = msg => {
-        console.debug(`Received message on internalPort`, appId, msg.data);
-        this.cw.mockFDC3Server?.receive(msg.data, connectionDetails.instanceId);
-      };
-
-      return connectionDetails.connectionId;
+      const ni = this.nextInstanceId++;
+      if (appId.includes('missing')) {
+        throw new Error(OpenError.AppNotFound);
+      } else {
+        const mc = new MessageChannel();
+        const internalPort = mc.port1;
+        const externalPort = mc.port2;
+
+        //   internalPort.name = 'internalPort-' + ni;
+        //   externalPort.name = 'externalPort-' + ni;
```

Comment:
Do these commented-out lines need to stay?

Reply:
nope